core/src/dird/dird_conf.h

```diff
@@ -250,8 +251,8 @@ class CatalogResource : public BareosResource {
   uint32_t mult_db_connections = 0; /**< Set if multiple connections wanted */
   bool disable_batch_insert
       = false;                /**< Set if batch inserts should be disabled */
-  bool try_reconnect = false; /**< Try to reconnect a database connection when
-                         its dropped */
+  bool try_reconnect = true;  /**< Try to reconnect a database connection when
+                          it is dropped */
```

Comment:
when the parameter from the configuration is removed, this should probably be removed, too. Which will make the compiler complain if you overlooked any used of `try_reconnect` somewhere.
It will at least require a change in the other call to `DbSqlGetPooledConnection()`.